ocaml/xapi-idl/cluster/cluster_interface.ml

```diff
@@ -15,12 +15,45 @@ type debug_info = string [@@deriving rpcty]
 (** Name of the cluster *)
 type cluster_name = string [@@deriving rpcty]
 
-(** An IPv4 address (a.b.c.d) *)
-type address = IPv4 of string [@@deriving rpcty]
+type ipv4v6 = IPv4 of string | IPv6 of string [@@deriving rpcty]
 
-let printaddr () = function IPv4 s -> Printf.sprintf "IPv4(%s)" s
+let string_of_ip = function IPv4 a | IPv6 a -> a
 
-let str_of_address address = match address with IPv4 a -> a
+type uuid_addr = {ip: ipv4v6; hostid: string; hostname: string}
```

Comment:
The naming of `uuid_addr ` could be slightly confusing:
1. Does it mean the uuid of addr?
2. If it means uuid and addr, whose uuid is it?

Reply:
it means it's a variant of the address type, namely a uuid address, which is the second point you mentioned. The uuid is the host uuid, stored inside hostid field.

Reply:
I agree this is not the best implementation, but in order to maintain backward compatibility, we cannot change the api interface too much otherwise they would be broken where there are different versions of the api, during, e.g. a pool upgrade

Reply:
I see. It's some kind of compromise.